### PR TITLE
[New Rule] AWS RDS DB Instance Made Public

### PR DESCRIPTION
**Pull request by imays11**
## Issues

- https://github.com/elastic/ia-trade-team/issues/346
## Summary
Identifies the creation or modification of an AWS RDS DB Instance to enable public access. DB instances may contain sensitive data that can be abused if shared with unauthorized accounts or made public. Adversaries may enable public access on a DB Instance to maintain persistence or evade defenses by bypassng access controls.

Screenshot 2024-06-29 at 3 31 44 AM

### CODE REVIEW

File: rules/integrations/aws/persistence_rds_instance_made_public.toml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+- **Enhance Monitoring and Alerts**: Adjust monitoring systems to alert on similar actions, especially those involving sensitive data or permissions.
+- **Audit Instances and Policies**: Conduct a comprehensive audit of all instances and associated policies to ensure they adhere to the principle of least privilege.
+- **Policy Update**: Review and possibly update your organization’s policies on DB instance access to tighten control and prevent unauthorized access.
+- **Incident Response**: If malicious intent is confirmed, consider it a data breach incident and initiate the incident response protocol. This includes further investigation, containment, and recovery.
+
+### Additional Information:
+
+For further guidance on managing DB backups and securing AWS environments, refer to the [AWS RDS documentation](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_RDS_Managing.html) and AWS best practices for security. Additionally, consult the following resources for specific details on DB snapshot security:
+- [AWS RDS ModifyDBInstance](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ModifyDBInstance.html)
+"""
```
Comment (imays11) [inline rules/integrations/aws/persistence_rds_instance_made_public.toml:55]:
```suggestion
For further guidance on managing DB instances and securing AWS environments, refer to the [AWS RDS documentation](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_RDS_Managing.html) and AWS best practices for security. Additionally, consult the following resources for specific details on DB instance security:
- [AWS RDS ModifyDBInstance](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ModifyDBInstance.html)
"""
```

File: rules/integrations/aws/persistence_rds_instance_made_public.toml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    - **Request and Response Parameters**: Check the `aws.cloudtrail.request_parameters` field in the CloudTrail event to identify the DB Instance Identifier and any other modifications made to the instance.
+- **Verify the Modified Instance**: Check the DB instance that was modified and its contents to determine the sensitivity of the data stored within it.
+- **Contextualize with Recent Changes**: Compare this modification event against recent changes in RDS DB or Cluster configurations and deployments. Look for any other recent permissions changes or unusual administrative actions.
+- **Correlate with Other Activities**: Search for related CloudTrail events before and after this change to see if the same actor or IP address engaged in other potentially suspicious activities.
+- **Interview Relevant Personnel**: If the modification was initiated by a user, verify the intent and authorization for this action with the person or team responsible for managing DB instances.
+### False Positive Analysis
+
+- **Legitimate Instance Modification**: Confirm if the Db instance modification aligns with legitimate tasks.
```
Comment (imays11) [inline rules/integrations/aws/persistence_rds_instance_made_public.toml:40]:
```suggestion
- **Legitimate Instance Modification**: Confirm if the DB instance modification aligns with legitimate tasks.
```

File: rules/integrations/aws/persistence_rds_instance_made_public.toml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+creation_date = "2024/06/29"
+integration = ["aws"]
+maturity = "production"
+updated_date = "2024/06/29"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation or modification of an AWS RDS DB Instance to enable public access. DB instances may contain sensitive data that can be abused if shared with unauthorized accounts or made public. Adversaries may enable public access on a DB Instance to maintain persistence or evade defenses by bypassng access controls.
+"""
```
Comment (terrancedejesus) [inline rules/integrations/aws/persistence_rds_instance_made_public.toml:11]:
```suggestion
Identifies the creation or modification of an AWS RDS DB instance to enable public access. DB instances may contain sensitive data that can be abused if shared with unauthorized accounts or made public. Adversaries may enable public access on a DB instance to maintain persistence or evade defenses by bypassing access controls.
"""
```

**Review by terrancedejesus — APPROVED** [review]
Great addition!

**Review by Aegrah — APPROVED** [review]

**Pull request merged**